pyccel/codegen/printing/cucode.py

```diff
@@ -131,4 +137,44 @@ def _print_ModuleHeader(self, expr):
                           global_variables,
                           function_declaration,
                           "#endif // {name.upper()}_H\n"))
+    def _print_Allocate(self, expr):
+        variable = expr.variable
+        if not isinstance(variable.class_type, CudaArrayType):
+            return super()._print_Allocate(expr)
+        shape = ", ".join(self._print(i) for i in expr.shape)
+        if isinstance(variable.class_type, CudaArrayType):
+            dtype = self.find_in_ndarray_type_registry(variable.dtype)
+        else:
```

Comment:
This if seems redundant with the one above that calls the C printer if the class type is not CudaArrayType